x-pack/legacy/plugins/advanced_ui_actions/public/customize_time_range_modal.tsx

```diff
@@ -0,0 +1,176 @@
+            timeRange: undefined,
+          },
+        },
+      },
+    });
+
+    this.props.onClose();
+  };
+
+  public render() {
+    return (
+      <React.Fragment>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle data-test-subj="customizePanelTitle">
+            {i18n.translate('xpack.advancedUiActions.customizeTimeRange.modal.headerTitle', {
+              defaultMessage: 'Customize panel time range',
+            })}
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+
+        <EuiModalBody>
+          {' '}
+          <EuiFormRow
+            label={i18n.translate(
+              'xpack.advancedUiActions.customizePanelTimeRange.modal.optionsMenuForm.panelTitleFormRowLabel',
+              {
+                defaultMessage: 'Time range',
+              }
+            )}
+          >
+            <EuiSuperDatePicker
+              start={this.state.timeRange ? this.state.timeRange.from : undefined}
+              end={this.state.timeRange ? this.state.timeRange.to : undefined}
+              isPaused={false}
+              onTimeChange={this.onTimeChange}
+              showUpdateButton={false}
+              dateFormat={this.props.dateFormat}
+            />
+          </EuiFormRow>
+        </EuiModalBody>
+        <EuiModalFooter>
+          <EuiFlexGroup justifyContent="spaceBetween">
+            <EuiFlexItem>
+              <EuiFlexGroup gutterSize="s">
+                <EuiButtonEmpty
+                  onClick={this.inheritFromParent}
+                  color="danger"
+                  data-test-subj="removePerPanelTimeRangeButton"
+                  disabled={this.state.inheritTimeRange}
+                >
+                  {' '}
+                  {i18n.translate(
+                    'xpack.advancedUiActions.customizePanelTimeRange.modal.removeButtonTitle',
+                    {
+                      defaultMessage: 'Remove',
+                    }
+                  )}
+                </EuiButtonEmpty>
+              </EuiFlexGroup>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiFlexGroup gutterSize="s">
+                <EuiButtonEmpty
+                  onClick={this.cancel}
+                  data-test-subj="cancelPerPanelTimeRangeButton"
+                >
+                  {' '}
+                  {i18n.translate(
+                    'xpack.advancedUiActions.customizePanelTimeRange.modal.cancelButtonTitle',
+                    {
+                      defaultMessage: 'Cancel',
+                    }
+                  )}
+                </EuiButtonEmpty>
+
+                <EuiButton
+                  data-test-subj="addPerPanelTimeRangeButton"
+                  onClick={this.addToPanel}
+                  fill
+                >
+                  {i18n.translate(
+                    'xpack.advancedUiActions.customizePanelTimeRange.modal.addToPanelButtonTitle',
+                    {
+                      defaultMessage: 'Add to panel',
```

Comment:
Can you make the text of this button dynamic to something like:

`this.state.inheritTimeRange ? 'Add to panel' : 'Update'`